test/labeling/test_convergence.py

```diff
@@ -0,0 +1,102 @@
+import unittest
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from snorkel.analysis.utils import set_seed
+from snorkel.labeling.apply.pandas import PandasLFApplier
+from snorkel.labeling.lf import LabelingFunction, labeling_function
+from snorkel.labeling.model.label_model import LabelModel
+from snorkel.labeling.preprocess import preprocessor
+from snorkel.types import DataPoint
+
+
+def create_data(n: int) -> pd.DataFrame:
+    """Create random pairs x1, x2 in [-1., 1.] with label x1 > x2 + 0.25."""
+    X = np.random.random((n, 2)) * 2 - 1
+    Y = (X[:, 0] > X[:, 1] + 0.25).astype(int) + 1
+
+    df = pd.DataFrame(
+        {"x0": np.random.randint(0, 1000, n), "x1": X[:, 0], "x2": X[:, 1], "y": Y}
+    )
+    return df
+
+
+def get_positive_labeling_function(divisor: int) -> LabelingFunction:
+    """Get LabelingFunction that abstains unless x0 is divisible by divisor."""
+
+    def f(x: DataPoint) -> int:
+        if x.x0 % divisor == 0 and x.x1 > x.x2 + 0.25:
+            return 2
+        else:
+            return 0
+
+    return LabelingFunction(f"lf_{divisor}", f)
```

Comment:
Nit: the names aren't unique anymore. Not required. But you might want `lf_pos_{divisor}`

Reply:
Done